cpp_test_suite/environment/setup.sh

```diff
@@ -2,36 +2,58 @@
 
 echo "Setup test environment"
 
-#run mysql-tango docker
-docker run --name mysql_db \
-    -e MYSQL_ROOT_PASSWORD=root \
-    -d tangocs/mysql:9.2.2 --sql-mode=""
-
-#run tango-cs docker
-CONTAINER=$(docker run --name tango_cs \
-    -p 10000:10000 \
-    -e TANGO_HOST=127.0.0.1:10000 \
-    -e MYSQL_HOST=mysql_db:3306 \
-    -e MYSQL_USER=tango \
-    -e MYSQL_PASSWORD=tango \
-    -e MYSQL_DATABASE=tango \
-    --link mysql_db:mysql_db \
-    -d tangocs/tango-cs:latest)
-
-echo "CONTAINER=$CONTAINER"
-
-IPADDR=$(docker inspect -f '{{ .NetworkSettings.IPAddress }}' $CONTAINER)
-#export TANGO_HOST
-TANGO_HOST=$IPADDR:10000
+function run_mysql_container {
+    docker run \
+        --rm \
+        --name $1 \
+        -e MYSQL_ROOT_PASSWORD=root \
+        -e MYSQL_INITDB_SKIP_TZINFO=1 \
+        -d \
+        tangocs/mysql:9.2.2 \
+        --sql-mode="" \
+        --innodb=OFF \
+        --default-storage-engine=MyISAM
+}
+
+function run_tango_container {
+    docker run \
+        --rm \
+        --name $1 \
+        -e TANGO_HOST=127.0.0.1:10000 \
+        -e MYSQL_HOST=$2:3306 \
+        -e MYSQL_USER=tango \
+        -e MYSQL_PASSWORD=tango \
+        -e MYSQL_DATABASE=tango \
+        --link $2:$2 \
+        -d \
+        tangocs/tango-cs:latest
+}
+
+run_mysql_container mysql_db
+run_mysql_container mysql_db2
+run_tango_container tango_cs mysql_db
+run_tango_container tango_cs2 mysql_db2
+
+IPADDR=$(docker inspect -f '{{ .NetworkSettings.IPAddress }}' tango_cs)
+IPADDR2=$(docker inspect -f '{{ .NetworkSettings.IPAddress }}' tango_cs2)
+
+export TANGO_HOST=$IPADDR:10000
+export TANGO_HOST2=$IPADDR2:10000
+
 echo "TANGO_HOST=$TANGO_HOST"
-export TANGO_HOST
 
 echo "Create tango_host file"
-echo "#!/bin/bash" > tango_host
-echo "export TANGO_HOST=$TANGO_HOST" >> tango_host
 
-echo "Wait till tango-cs is online"
-#TODO notification?
-sleep 30
+cat << EOF > tango_host
+#!/bin/bash
+export TANGO_HOST=$TANGO_HOST
+export TANGO_HOST2=$TANGO_HOST2
+EOF
 
-#ctest
\ No newline at end of file
+echo "Wait till tango-cs is online"
+if hash tango_admin 2>/dev/null; then
+    tango_admin --ping-database 30
```

Comment:
Great improvement!

Reply:
This is probably 3rd PR where I'm adding this. Maybe finally we will get it on master ;)

Reply:
@mliszcz Just push the change out into a separate PR so we can get the easy changes merged fast.

Reply:
@t-b here is a simple enough PR with just this kind of changes: https://github.com/tango-controls/cppTango/pull/557
But let's ignore that one for now and focus on current one.